infor-design/enterprise
# 5062/5063 - Time/Datepicker icon placement improvements [v4.50.x]

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR resolves the following:
- Changes from #5030 applied a trigger button in places where only an icon used to exist.  On the smaller-sized datagrid filter fields, these buttons weren't properly aligned anymore.  This is now fixed.
- Changes in #5030 accidentally altered the behavior of readonly Lookup fields, which should still allow for the user to enter a lookup field by clicking the icon.  The behavior of lookup buttons in this PR has been altered to allow the button to be clicked, but not focused by keyboard.
- Changes from #5022 improved the alignment of the close button on Searchfield inputs, but the alignment was altered specifically on non-collapsible Searchfields inside of Lookups.  This alignment is now fixed.

**Related github/jira issue (required)**:
- Related to #5022 
- Related to #5030
- Closes #5062 
- Closes #5063
- Closes #5087

**Steps necessary to review your pull request (required)**:
Pull, build, run

_Test #5062:_
- open http://localhost:4000/components/lookup/example-not-tabbable.html
- click the Readonly Lookup field's trigger button.  It should activate the Lookup as expected

_Test #5063:_
- open http://localhost:4000/components/datagrid/example-filter.html
- Using the "..." button, examine the datagrid headers' filter fields, specifically the Lookup, Timepicker, and Datepicker, on the "Short" and "Extra Short" grid sizes.  The icons in these filter fields should be center-aligned vertically and close to the right-most edge.
- Open Datepickers within the grid and ensure when focusing the cell, their icons do not disappear or become misaligned.  Also ensure the Popup opens in the correct place.
- Test Timepicker Fields: http://localhost:4000/components/datagrid/test-editable-time.html
- Test Lookup Fields: http://localhost:4000/components/datagrid/test-editable-lookup.html
- Test RTL: http://localhost:4000/components/datagrid/example-filter.html?locale=ar-EG
- Test Classic theme: http://localhost:4000/components/datagrid/example-filter.html?theme=classic&mode=light
- Test Classic theme + RTL: http://localhost:4000/components/datagrid/example-filter.html?theme=classic&mode=light&locale=ar-EG


_Test "X" misalignment:_
- open http://localhost:4000/components/datagrid/test-lookup-filter.html.  Examine the lookup field (labelled "Product ID).  The X button should be center-aligned vertically and close to the right-most edge.
- open http://localhost:4000/components/lookup/example-not-tabbable.html and open the Lookup.  Enter some text in the Searchfield.  When the "X" appears, it should be center-aligned vertically and close to the right-most edge.

**Included in this Pull Request**:
